### PR TITLE
airbyte-cdk [python]: re-enable tests in CI

### PR DESCRIPTION
**Pull request by alafanechere**
`gradle.yml` is not running the python cdk test anymore as the python cdk is now an independant (and soon retired) gradle project. This PR introduces a new workflow which should specifically run `gradle check` on the python cdk.

## TODO:
* We can't test this workflow until it has reached master, so merge and test...
* Make the "Python CDK Tests" a required check if we want to disallow merges when they fail (GH branch protection rules settings)

### CODE REVIEW

**Comment by alafanechere** [comment]
* **#35560** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/35560?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alafanechere and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/python_cdk_tests.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: cd airbyte-cdk/python
+        run: cd airbyte-cdk/python
+      - name: Run Gradle Check
+        uses: burrunan/gradle-cache-action@v1
+        env:
+          CI: true
+        with:
+          job-id: gradle-check
```
Comment (postamar) [inline .github/workflows/python_cdk_tests.yml:70]:
```suggestion
          job-id: python-cdk-check
```

File: .github/workflows/python_cdk_tests.yml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        run: python3 -m pip install virtualenv --user
+      - name: Docker login
+        # Some tests use testcontainers which pull images from DockerHub.
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: cd airbyte-cdk/python
+        run: cd airbyte-cdk/python
+      - name: Run Gradle Check
```
Comment (postamar) [inline .github/workflows/python_cdk_tests.yml:65]:
```suggestion
      - name: Run Gradle Check on Python CDK
```

Comment (postamar) [reply .github/workflows/python_cdk_tests.yml:65]: You need to name this something else in order to make this a required branch check.

File: .github/workflows/python_cdk_tests.yml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      - name: Install Pip
+        run: curl -fsSL https://bootstrap.pypa.io/get-pip.py | python3
+      - name: Install Pyenv
+        run: python3 -m pip install virtualenv --user
+      - name: Docker login
+        # Some tests use testcontainers which pull images from DockerHub.
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
```
Comment (postamar) [inline .github/workflows/python_cdk_tests.yml:62]:
```suggestion
```

We don't involve docker in the python CDK tests, do we?

File: .github/workflows/python_cdk_tests.yml [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        uses: burrunan/gradle-cache-action@v1
+        env:
+          CI: true
+        with:
+          job-id: gradle-check
+          read-only: ${{ github.ref != 'refs/heads/master' }}
+          gradle-distribution-sha-256-sum-warning: false
+          concurrent: true
+          # TODO: be able to remove the skipSlowTests property
+          arguments: --scan check -DskipSlowTests=true
```
Comment (postamar) [inline .github/workflows/python_cdk_tests.yml:75]:
```suggestion
          arguments: --scan build
```

**Review by postamar — CHANGES_REQUESTED** [review]

**Review by girarda — APPROVED** [review]
looks good in the sense this looks very similar to the java workflow. 🚢  and 🙏

**Review by postamar — APPROVED** [review]
I'll get this merged and I'll make sure it works. It's late for @alafanechere , I'll take over.

**Pull request merged**